src/query_plan/mod.rs

```diff
@@ -1 +1,210 @@
+use apollo_compiler::executable::{Field, InlineFragment, OperationType, Selection, SelectionSet};
+use apollo_compiler::{ExecutableDocument, Node, NodeStr};
+
 pub mod operation;
+pub mod query_planner;
+
+pub struct QueryPlan {
+    node: Option<TopLevelPlanNode>,
+}
+
+pub enum TopLevelPlanNode {
+    Subscription(Node<SubscriptionNode>),
```

Comment:
`apollo_compiler::Node` is primarily an `Arc` with source location information, is source location relevant here? If not, this probably should be either an Arc or just nothing (`Subscription(SubscriptionNode)`)

Reply:
Good callout, there's no relevant source information. Will use `Arc` instead.

Reply:
Will also remove `Arc` from `TopLevelPlanNode`.